### ISSUE TITLE
Struggling to render a .obj file from code

### ISSUE THREAD

**Issue by yvessavoy** [amethyst/amethyst#2104]
I tried rendering a simple .obj-file from code directly, but I cannot get it working. I was able to render a Sphere with the following snippet:
```rust
let mesh = world.exec(|loader: AssetLoaderSystemData<'_, Mesh> | {
        loader.load_from_data(
            Shape::Sphere(100, 100)
                .generate::<(Vec<Position>, Vec<Normal>, Vec<Tangent>, Vec<TexCoord>)>(None)
                .into(),
            (),
        )
    });
```

But whenever I try replacing the Sphere with an object-file, Amethyst panics with the following error:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Incompatible { not_found: VertexFormat { stride: 16, attributes: [Attribute { uuid: AttrUuid(4), element: Element { format: Rgba32Sfloat, offset: 0 }, index: 0, name: "tangent" }] }, in_formats: [VertexFormat { stride: 8, attributes: [Attribute { uuid: AttrUuid(3), element: Element { format: Rg32Sfloat, offset: 0 }, index: 0, name: "tex_coord" }] }, VertexFormat { stride: 12, attributes: [Attribute { uuid: AttrUuid(2), element: Element { format: Rgb32Sfloat, offset: 0 }, index: 0, name: "normal" }] }, VertexFormat { stride: 12, attributes: [Attribute { uuid: AttrUuid(1), element: Element { format: Rgb32Sfloat, offset: 0 }, index: 0, name: "position" }] }] }', src\libcore\result.rs:1165:5
```

Can someone explain what is happening here? I see that when rendering the Sphere, I provided Position, Normal, etc to the loader, I did not do this when loading the object file. Does this cause the issue?

Full code:
```rust
use amethyst::{
    assets::AssetLoaderSystemData,
    core::{TransformBundle, Transform},
    prelude::*,
    renderer::{
        plugins::{
            RenderPbr3D,
            RenderToWindow
        },
        light::{
            Light,
            PointLight,
        },
        rendy::{
            mesh::{
                Normal,
                Position,
                Tangent,
                TexCoord
            },
        },
        shape::Shape,
        mtl::{Material, MaterialDefaults},
        types::DefaultBackend,
        RenderingBundle,
        Camera,
        palette::rgb::Rgb,
        Mesh,
        formats::mesh::ObjFormat,
    },
    utils::application_root_dir,
};

struct GameState;

impl SimpleState for GameState {
    fn on_start(&mut self, state_data: StateData<'_, GameData<'_, '_>>) {
        initialize_camera(state_data.world);
        initialize_sphere(state_data.world);
        initialize_light(state_data.world);
    }
}

fn initialize_sphere(world: &mut World) {
    let mesh = world.exec(|loader: AssetLoaderSystemData<'_, Mesh> | {
        loader.load("gas-station.obj", ObjFormat, ())
    });

    let material_defaults = world.read_resource::<MaterialDefaults>().0.clone();
    let material = world.exec(|loader: AssetLoaderSystemData<'_, Material> | {
        loader.load_from_data(
            Material {
                ..material_defaults
            },
            (),
        )
    });

    let mut transform = Transform::default();
    transform.set_translation_xyz(0.0, 0.0, 0.0);

    world.create_entity()
        .with(mesh)
        .with(material)
        .with(transform)
        .build();
}

fn initialize_light(world: &mut World) {
    let light: Light = PointLight {
        intensity: 10.0,
        color: Rgb::new(1.0, 1.0, 1.0),
        ..PointLight::default()
    }.into();

    let mut transform = Transform::default();
    transform.set_translation_xyz(5.0, 5.0, 20.0);

    world
        .create_entity()
        .with(light)
        .with(transform)
        .build();
}

fn initialize_camera(world: &mut World) {
    let mut transform = Transform::default();
    transform.set_translation_xyz(0.0, 0.0, 10.0);

    world.create_entity()
        .with(Camera::standard_3d(500.0, 500.0))
        .with(transform)
        .build();
}

fn main() -> amethyst::Result<()> {
    amethyst::start_logger(Default::default());

    let app_root = application_root_dir()?;
    let display_config_path = app_root.join("config/display.ron");
    let assets_dir = app_root.join("assets/");

    let game_data = GameDataBuilder::default()
        .with_bundle(TransformBundle::new())?
        .with_bundle(
            RenderingBundle::<DefaultBackend>::new()
                .with_plugin(
                    RenderToWindow::from_config_path(display_config_path)
                        .with_clear([0.529, 0.808, 0.98, 1.0]),
                )
                .with_plugin(RenderPbr3D::default())
        )?;

    let mut game = Application::new(assets_dir, GameState, game_data)?;
    game.run();

    Ok(())
}
```

Thanks & Cheers

**Comment by jojolepro** [comment]
I would suggest asking your question on the Amethyst Discord server or on the forum for a faster reply.

**Comment by GrantMoyer** [comment]
From Valkum in the discord server:

> Do you use the Pbr pass? The obj load does not calculate the tangents currently and the PBR pass needs tangents. Try the Shaded pass for use with obj files.

**Comment by jojolepro** [comment]
tldr; you should use a flat rendering pass, because the pbr pass uses data which is not present in .obj files.
